Microsoft.Azure.Cosmos/src/Resource/Settings/DedicatedGatewayRequestOptions.cs

```diff
@@ -21,17 +21,25 @@ class DedicatedGatewayRequestOptions
         /// <summary> 
         /// Gets or sets the staleness value associated with the request in the Azure CosmosDB service. 
         /// </summary> 
+        /// <value>Default value is null.</value> 
         /// <remarks> 
         /// For requests where the <see cref="ConsistencyLevel"/> is <see cref="ConsistencyLevel.Eventual"/>, responses from the integrated cache are guaranteed to be no staler than value indicated by this MaxIntegratedCacheStaleness. 
-        /// Value defaults to null. 
+        /// Cache Staleness is supported in milliseconds granularity. Anything smaller than milliseconds will be ignored.
         /// </remarks> 
         public TimeSpan? MaxIntegratedCacheStaleness { get; set; }
 
         internal static void PopulateMaxIntegratedCacheStalenessOption(DedicatedGatewayRequestOptions dedicatedGatewayRequestOptions, RequestMessage request)
         {
             if (dedicatedGatewayRequestOptions?.MaxIntegratedCacheStaleness != null)
             {
-                request.Headers.Set(HttpConstants.HttpHeaders.DedicatedGatewayPerRequestCacheStaleness, dedicatedGatewayRequestOptions.MaxIntegratedCacheStaleness.Value.ToString("c", CultureInfo.InvariantCulture));
+                double cacheStalenessInMilliseconds = (double)dedicatedGatewayRequestOptions?.MaxIntegratedCacheStaleness.Value.TotalMilliseconds; 
```

Comment:
Already checked for null. No need for it.
```suggestion
                double cacheStalenessInMilliseconds = (double)dedicatedGatewayRequestOptions.MaxIntegratedCacheStaleness.Value.TotalMilliseconds; 
```